yarn-project/aztec/src/cli/cmds/start_pxe.ts

```diff
@@ -40,43 +47,44 @@ export async function addPXE(
   userLog: LogFn,
   deps: { node?: AztecNode } = {},
 ) {
-  const pxeCliOptions = parseModuleOptions(options.pxe);
-  const pxeConfig = mergeEnvVarsAndCliOptions<CliPXEOptions>(getCliPXEOptions(), pxeCliOptions);
+  // const pxeCliOptions = parseModuleOptions(options.pxe);
+  // const pxeConfig = mergeEnvVarsAndCliOptions<CliPXEOptions>(getCliPXEOptions(), pxeCliOptions);
+  const pxeConfig = extractRelevantOptions<CliPXEOptions>(options, pxeCliConfigMappings);
   let nodeUrl;
-  if (pxeCliOptions.network) {
-    if (isValidNetwork(pxeCliOptions.network)) {
-      if (!pxeCliOptions.apiKey) {
-        userLog(`API Key is required to connect to ${pxeCliOptions.network}`);
+  if (pxeConfig.network) {
+    if (isValidNetwork(pxeConfig.network)) {
+      if (!pxeConfig.apiKey) {
+        userLog(`API Key is required to connect to ${pxeConfig.network}`);
         process.exit(1);
       }
-      nodeUrl = `https://api.aztec.network/${pxeCliOptions.network}/aztec-node-1/${pxeCliOptions.apiKey}`;
+      nodeUrl = `https://api.aztec.network/${pxeConfig.network}/aztec-node-1/${pxeConfig.apiKey}`;
     } else {
-      userLog(`Network ${pxeCliOptions.network} is not supported`);
+      userLog(`Network ${pxeConfig.network} is not supported`);
       process.exit(1);
     }
   } else {
-    nodeUrl = pxeCliOptions.nodeUrl ?? process.env.AZTEC_NODE_URL;
+    nodeUrl = pxeConfig.nodeUrl;
```

Comment:
IMO nodeUrl should take precedence over the default one derived from the network